sdk/src/types/positions/contracts/pool_manager.rs

```diff
@@ -250,14 +266,130 @@ pub async fn pool_manager_pool_liquidity<F: StorageSlotFetcher>(
     Ok(liquidity)
 }
 
+pub async fn pool_manager_load_tick_map<F: StorageSlotFetcher>(
+    slot_fetcher: &F,
+    pool_manager_address: Address,
+    block_number: Option<u64>,
+    pool_id: PoolId,
+    tick_spacing: I24,
+    start_tick: Option<I24>,
+    end_tick: Option<I24>,
+    load_buffer: Option<usize>
+) -> eyre::Result<HashMap<I24, TickData>> {
+    let start_tick = start_tick
+        .map(|t| normalize_tick(t, tick_spacing))
+        .unwrap_or(I24::unchecked_from(-887272));
+    let end_tick = end_tick
+        .map(|t| normalize_tick(t, tick_spacing))
+        .unwrap_or(I24::unchecked_from(887272));
+
+    let mut tick_data_loading_stream =
+        futures::stream::iter(start_tick.as_i32()..=end_tick.as_i32())
```

Comment:
This is degen as we will attempt to load ticks that aren't aligned and thus will be zero. 
```
 (start_tick..=end_tick).step_by(self.tick_spacing)
 ```